sklearn/svm/liblinear.pyx

```diff
@@ -14,7 +14,8 @@ np.import_array()
 def train_wrap(X, np.ndarray[np.float64_t,   ndim=1, mode='c'] Y,
                bint is_sparse, int solver_type, double eps, double bias,
                double C, np.ndarray[np.float64_t, ndim=1] class_weight,
-               int max_iter, unsigned random_seed, double epsilon):
+               int max_iter, unsigned random_seed, double epsilon,
+               np.ndarray[np.float64_t, ndim=1] sample_weight):
```

Comment:
It can be C or F but not specifying it inserts code for arbitrary strides. E.g., `np.zeros(4)[::2]` is neither F- nor C-contiguous.

In my experience, specifying the mode doesn't buy you a lot of performance. The actual access pattern is more important than the few extraneous instructions. YMMV, though.